### ISSUE TITLE
Microsoft Bot Application Template not appear on Visual Studio 2017

### ISSUE THREAD

**Issue by JeevaJose** [Microsoft/botbuilder-dotnet#1101]
I have successfully installed the latest version of C# Bot template (version 4.1.5.43). after that it is not appear on new project template on Visual Studio 2017. My visual  studio installation is up to date but don’t know what is blocking to load the bot template project.

**Comment by JasonSowers** [comment]
How did you install the template? 

Here are the steps I would use:

- In visual studio go to tool>Extensions and Updates
![image](https://user-images.githubusercontent.com/6301037/48367791-996e5e80-e666-11e8-8081-fde3b59c22cd.png)

- Select "online" and search for "Bot Builder"
![image](https://user-images.githubusercontent.com/6301037/48367874-e94d2580-e666-11e8-9165-ae8b98f6d6e4.png)

- Follow the instructions and once you restart visual studio and do file>new to make a new project it should look like this
![image](https://user-images.githubusercontent.com/6301037/48367951-231e2c00-e667-11e8-946c-bf6d69757d0a.png)


Let me know if this helps or not, we can look into this further is needed.

**Comment by JeevaJose** [comment]
I had the same template which you have been mentioned on above discussion but it was quite old and Microsoft.Bot.Builder version is 4.0.6., The old template was working as required. I have installed the latest version (4.1.5.43) of the template which I have taken from following link:
https://botbuilder.myget.org/feed/aitemplates/package/vsix/BotBuilderV4.fbe0fc50-a6f1-4500-82a2-189314b7bea2

still I am facing the same issue.

**Comment by JasonSowers** [comment]
@JeevaJose and @bryan7ran thanks for reporting this, I am able to repro and we are looking into a fix.

**Comment by Zerryth** [comment]
@JeevaJose just dug down into the templates. It looks like the most recent one that works as intended is version 4.1.5.41

Versions more recent than that are all facing the same bug that you have reported

Searching for a fix

**Comment by Zerryth** [comment]
So downloading the [source](https://github.com/Microsoft/BotBuilder-Samples/tree/master/generators/vsix-vs-win/BotBuilderVSIX-V4) and building the `BotBuilderTemplaterVSIX` project, and installing the VSIX locally

Tools > Extensions and Updates
![image](https://user-images.githubusercontent.com/35248895/49050260-61ebd000-f197-11e8-9140-766cf37c5a77.png)


![image](https://user-images.githubusercontent.com/35248895/49050137-d1ad8b00-f196-11e8-8d72-b4c0b70e8451.png)

we see that see `.vsix` is only 55 KB (rather small).

Then when you try to build a new project, you'll notice that the Bot Builder V4 **Template is still not available**, even though the extension is installed

![image](https://user-images.githubusercontent.com/35248895/49050490-71b7e400-f198-11e8-8e1b-778d006f5d6e.png)

___

### manually adding the templates

Right click on `Bot Framework.zip` to include in project > Build Action: Content > Include in VSIX: True

![image](https://user-images.githubusercontent.com/35248895/49050611-0589b000-f199-11e8-98ac-0d28701933b3.png)

`.vsix` file is now larger with the templates added (304 KB)
![image](https://user-images.githubusercontent.com/35248895/49050643-2b16b980-f199-11e8-87e0-0eabbb91d7c3.png)

Uninstalling the 55 KB extension, then installing the templates-included 304 KB `.vsix`...

now when you go to create a new project, **Bot Builder V4 template is now available**

![image](https://user-images.githubusercontent.com/35248895/49050908-78dff180-f19a-11e8-9af4-d47b15849fb6.png)
___

#### TL;DR the `.vsix` for the v4 templates currently are not including the `Bot Framework.zip`

It looks like the culprit is the commit [here](https://github.com/Microsoft/BotBuilder-Samples/commit/ab55afa098ed67057762175ceff7cfa136d2fd2e#diff-1a46c55fd0d4616da706ab9721d6816b) 
 **It's not enough to add the event to prebuild the `Bot Framework.zip`--it must also be included in the solution and included in the VSIX**

____

I'll make a PR with fix tonight, after commuting home :)

**Comment by Zerryth** [comment]
Track https://github.com/Microsoft/BotBuilder-Samples/pull/992 for revamping of VSIX that folds in a fix for the template issues experienced in this thread

Will fold in changes to make templates appear. 

See previous comment if you need a more immediate work around, before the larger set of changes are pushed.
